lib/core/src/Cardano/Wallet.hs

```diff
@@ -969,6 +985,119 @@ listTransactions ctx wid mStart mEnd order = db & \DBLayer{..} -> do
         -- the start of a slot and its end could be a valid candidate.
         txTime = slotStartTime sp
 
+createCert
+    :: forall ctx s t k.
+        ( HasTransactionLayer t k ctx
+        , HasLogger ctx
+        , HasDBLayer s k ctx
+        )
+    => ctx
+    -> WalletId
+    -> ExceptT ErrCreateCert IO CoinSelection
+createCert ctx wid = do
+    (wal, _, pending) <- withExceptT ErrCreateCertNoSuchWallet $
+        readWallet @ctx @s @k ctx wid
+    let bp = blockchainParameters wal
+    let utxo = availableUTxO @s pending wal
+    let sel = CoinSelection [] [] []
+    withExceptT ErrCreateCertFee $ do
+        debug tr "Coins selected for delegation certificate after fee adjustment"
+            =<< adjustForFee (feeOpts tl computeCertFee (bp ^. #getFeePolicy)) utxo sel
+  where
+    tl = ctx ^. transactionLayer @t @k
+    tr = ctx ^. logger
+
+signCert
+    :: forall ctx s t k.
+        ( HasTransactionLayer t k ctx
+        , HasDBLayer s k ctx
+        , Show s
+        , NFData s
+        , IsOwned s k
+        , GenChange s
+        , HardDerivation k
+        , AddressIndexDerivationType k ~ 'Soft
+        )
+    => ctx
+    -> WalletId
+    -> ArgGenChange s
+    -> Passphrase "encryption"
+    -> CoinSelection
+    -> PoolId
+    -> ExceptT ErrSignCert IO (Tx, TxMeta, UTCTime, [TxWitness])
+signCert ctx wid argGenChange pwd
+    (CoinSelection ins outs chgs) _poolId = db & \DBLayer{..} -> do
+    withRootKey @_ @s ctx wid pwd ErrSignCertWithRootKey $ \xprv -> do
+        mapExceptT atomically $ do
+            cp <- withExceptT ErrSignCertNoSuchWallet $ withNoSuchWallet wid $
+                readCheckpoint (PrimaryKey wid)
+            let (changeOuts, s') = flip runState (getState cp) $
+                    forM chgs $ \c -> do
+                        addr <- state (genChange argGenChange)
+                        return $ TxOut addr c
+            allShuffledOuts <- liftIO $ shuffle (outs ++ changeOuts)
+
+            --deriving chimeric account
+            let accPrv = deriveAccountPrivateKey @k pwd xprv minBound
+            let _addrPrv =
+                    deriveAddressPrivateKey pwd accPrv MutableAccount minBound
+
+            {--
+            let keyFrom = isOwned (getState cp) (xprv, pwd)
+            (tx, wit) <- either
+                (throwE . ErrSignCert)
+                pure
+                (mkCertificateTx tl keyFrom (poolId, addrPrv, pwd) ins allShuffledOuts)
+            --}
```

Comment:
Generally speaking, I'd suggest putting an explicit `error "not implemented"` and avoid writing functions that:

1. Compiles
2. Raises no runtime exception
3. Are not correct

In this case, I am pretty sure that we won't forgot to update it, but, as a habit, it's good to make sure that "if it compiles, it works".  If it doesn't work, either make sure it doesn't compile, or make sure it fails loudly :)

Reply:
my intention was to wait for mkCertificateTx to be merged, then rebase this PR and only when the functionalities work as we expect and all integration tests pass, merge this PR (according to the rule, that only well tested changes can be merged, no partial, untested code in master etc)